packages/core/src/shipping/strategies/paypal-commerce/paypal-commerce-accelerated-checkout-shipping-strategy.ts

```diff
@@ -0,0 +1,175 @@
+    PayPalCommerceAcceleratedCheckoutUtils,
+    PayPalCommerceConnectAuthenticationState,
+    PayPalCommerceConnectStylesOption,
+} from '@bigcommerce/checkout-sdk/paypal-commerce-utils';
+
+import { AddressRequestBody } from '../../../address';
+import { BillingAddressActionCreator } from '../../../billing';
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { InvalidArgumentError } from '../../../common/error/errors';
+import { PaymentMethodActionCreator } from '../../../payment';
+import { PaymentProviderCustomerActionCreator } from '../../../payment-provider-customer';
+import ConsignmentActionCreator from '../../consignment-action-creator';
+import { ShippingRequestOptions } from '../../shipping-request-options';
+import ShippingStrategy from '../shipping-strategy';
+
+import PayPalCommerceAcceleratedCheckoutShippingInitializeOptions from './paypal-commerce-accelerated-checkout-shipping-initialization-options';
+
+export default class PayPalCommerceAcceleratedCheckoutShippingStrategy implements ShippingStrategy {
+    constructor(
+        private _store: CheckoutStore,
+        private _billingAddressActionCreator: BillingAddressActionCreator,
+        private _consignmentActionCreator: ConsignmentActionCreator,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _paymentProviderCustomerActionCreator: PaymentProviderCustomerActionCreator,
+        private _paypalCommerceAcceleratedCheckoutUtils: PayPalCommerceAcceleratedCheckoutUtils,
+    ) {}
+
+    updateAddress(
+        address: AddressRequestBody,
+        options?: ShippingRequestOptions,
+    ): Promise<InternalCheckoutSelectors> {
+        return this._store.dispatch(this._consignmentActionCreator.updateAddress(address, options));
+    }
+
+    selectOption(
+        optionId: string,
+        options?: ShippingRequestOptions,
+    ): Promise<InternalCheckoutSelectors> {
+        return this._store.dispatch(
+            this._consignmentActionCreator.selectShippingOption(optionId, options),
+        );
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        return Promise.resolve(this._store.getState());
+    }
+
+    async initialize(
+        options: PayPalCommerceAcceleratedCheckoutShippingInitializeOptions,
+    ): Promise<InternalCheckoutSelectors> {
+        const { methodId, styles } = options || {};
+
+        if (!methodId) {
+            throw new InvalidArgumentError(
+                'Unable to proceed because "methodId" argument is not provided.',
+            );
+        }
+
+        // Should run Authentication flow
+        if (!this._shouldRunAuthenticationFlow()) {
+            return Promise.resolve(this._store.getState());
+        }
+
+        // Load payment method
+        // Load run auth flow
+        try {
+            await this._store.dispatch(
+                this._paymentMethodActionCreator.loadPaymentMethod(methodId),
```

Comment:
Do you need to call `loadPaymentMethod` here? This will trigger an API call which can slow down the render. In this case the `getPaymentMethod` should work 🤔

Reply:
Let me double check this part

Reply:
We should load payment method to get access to payment method data. PP AXO Authentication flow should run only when customer lands on Shipping step (after cart update or reloads a page after passing customer step), so in this case we will not have payment method to get it from state

Reply:
Is this only fired for logged in shopper or is fired for guest shopper as well? I ask since for guest shopper we load the payment method in customer step and set that in state. So in that case we can read from state, and load method if not present in state.

Reply:
It fires for customer who was authenticated in the same session before. So this is rather an edge case to get and autofill checkout form with PP Connect data